### PR TITLE
QPPCT-603 measure id suggestions

### PR DESCRIPTION
**Pull request by clydet**
### Information
- QPPCT-603

### Changes proposed in this PR:
- Introduces a scheme where suggestions may be provided for errant measure ids

### Checklist 
- [x] All JUnit tests pass (`mvn clean verify`).
- [x] New unit tests written to cover new functionality.
- [ ] Added and updated JavaDocs for non-test classes and methods.
- [ ] No local design debt. Do you feel that something is "ugly" after your changes?
- [ ] Updated documentation (`README.md`, etc.) depending if the changes require it.

### CODE REVIEW

File: commons/src/main/java/gov/cms/qpp/conversion/model/validation/MeasureConfigs.java [hunk 1]
```diff
@@ -49,6 +62,22 @@ private static void initMeasureConfigs() {
 						config.getCpcPlusGroup(), key -> new ArrayList<>()).add(config));
 	}
 
+	private static void initSpellChecker() {
+		File indexDir = new File(EnvironmentHelper.getOrDefault(MEASURES_INDEX_DIR, INDEX_DEFAULT));
```
Comment (omnivion) [inline commons/src/main/java/gov/cms/qpp/conversion/model/validation/MeasureConfigs.java]: why make this a file when it can just be a path?

**Comment by omnivion** [comment]
Really like the lucene usage, forgot that was a thing

File: commons/src/main/java/gov/cms/qpp/conversion/model/validation/MeasureConfigs.java [hunk 2]
```diff
@@ -2,26 +2,38 @@
...
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
 public class MeasureConfigs {
+	static final String MEASURES_INDEX_DIR = "MEASURES_INDEX_DIR";
```
Comment (halprin) [inline commons/src/main/java/gov/cms/qpp/conversion/model/validation/MeasureConfigs.java]: Can you add this to the `.env` file with a description?

File: converter/src/test/java/gov/cms/qpp/acceptance/cpc/CpcPlusAcceptanceTest.java [hunk 3]
```diff
@@ -35,15 +36,15 @@
 	private static Map<String, CPCAcceptanceFixture> fixtureValues;
 
 	@BeforeAll
-	static void initMockApmIds() throws IOException {
+	static void initMockApmIds() throws IOException, NoSuchMethodException, IllegalAccessException, InvocationTargetException {
```
[the remark this answers is not in the document]
Comment (clydet) [reply converter/src/test/java/gov/cms/qpp/acceptance/cpc/CpcPlusAcceptanceTest.java]: no, holdover from a refactor

File: commons/src/main/java/gov/cms/qpp/conversion/model/validation/MeasureConfigs.java [hunk 4]
```diff
@@ -2,26 +2,38 @@
...
 	private static final Logger DEV_LOG = LoggerFactory.getLogger(MeasureConfigs.class);
 	public static final String DEFAULT_MEASURE_DATA_FILE_NAME = "measures-data.json";
+	static final int SUGGESTION_COUNT = 3;
+	private static final String INDEX_DEFAULT = "/usr/src/run/measures_index";
```
[the remark this answers is not in the document]
Comment (clydet) [reply commons/src/main/java/gov/cms/qpp/conversion/model/validation/MeasureConfigs.java]: No, lucene doesn't come with a `org.apache.lucene.store.Directory` implementation that can read an index from within a jar. Thinking the compression compromises with the index files.

File: commons/src/main/java/gov/cms/qpp/conversion/model/validation/MeasureConfigs.java (hunk 2 above)
Comment (clydet) [reply commons/src/main/java/gov/cms/qpp/conversion/model/validation/MeasureConfigs.java]: sure

**Comment by clydet** [comment]
Don't you guys?

**Comment by clydet** [comment]
@halprin I was trying to personify the PR.

**Comment by halprin** [comment]
Oh, lol.

**Pull request merged**